cromshell

```diff
@@ -427,6 +456,42 @@ function populateLastWorkflowId()
   WORKFLOW_ID=$( tail -n1 ${CROMWELL_SUBMISSIONS_FILE} | awk '{print $3}' )
 }
 
+function alias_workflow() 
+{
+	local alias_name=${2}
+
+	# Check that we have an alias and workflow defined:
+	if [[ ${#alias_name} -eq 0 ]] || [[ ${#2} -eq 0 ]] ; then
+		error "ERROR: Cannot process alias name."
+		error "       Please specify both a workflow ID and an alias."
+		return 1
+	fi
+
+	# Check if the alias already exists:
+	awk '{print $6}' ${CROMWELL_SUBMISSIONS_FILE} | grep -v '^[ \t]*$' | grep -q "^${alias_name}$"
+	local r=$?
+	if [[ $r -eq 0 ]] ; then
+		error "ERROR: Alias already exists: ${alias_name}"
+		error "       Please choose another alias."
+		return 1
+	fi
+
+	# Get the workflow ID:
+	populateWorkflowIdAndServerUrl ${1}
+
+	error "Aliasing ${WORKFLOW_ID} -> ${alias_name}"
+	cp ${CROMWELL_SUBMISSIONS_FILE} ${CROMWELL_SUBMISSIONS_FILE}.bak
```

Comment:
do you want to remove the .bak file after?

Reply:
should we be worried about multiple cromshell commands being run in parallel?

Reply:
No - there's a sed command elsewhere in the script that saves to this same backup file.  I also like keeping a backup file "just in case".  In most cases the file will be small enough not to matter.

So far we have not put multiprocessing/multithreadding as a requirement in cromshell.  If we want to, that's fine, but we'll need to change a lot more stuff too.